src/librustc/session/mod.rs

```diff
@@ -1128,14 +1128,21 @@ pub fn build_session_(
         // positives, or in other words we try to execute this before we open
         // any file descriptors ourselves.
         //
+        // Pick a "reasonable maximum" if we don't otherwise have
+        // a jobserver in our environment, capping out at number of cpus so we
+        // don't take everything down by hogging the process run queue.
+        //
         // Also note that we stick this in a global because there could be
         // multiple `Session` instances in this process, and the jobserver is
         // per-process.
-        jobserver_from_env: unsafe {
-            static mut GLOBAL_JOBSERVER: *mut Option<Client> = 0 as *mut _;
+        jobserver: unsafe {
+            static mut GLOBAL_JOBSERVER: *mut Client = 0 as *mut _;
             static INIT: std::sync::Once = std::sync::ONCE_INIT;
             INIT.call_once(|| {
-                GLOBAL_JOBSERVER = Box::into_raw(Box::new(Client::from_env()));
+                let client = Client::from_env().unwrap_or_else(|| {
+                    Client::new(::num_cpus::get()).expect("failed to create jobserver")
```

Comment:
@andjo403, given @alexcrichton's comment below, could you change this 32? Most of the time there'll be a jobserver around anyway.

Reply:
@michaelwoerister have updated now and added a comment about why there is a fixed value.
Thanks for the feedback